File: generated-code/swift/codecraft/FileReadWrite/Sources/TransGenTest/Model/BuildAction.swift
/// Build action
public struct BuildAction {
    /// Type of an entity to build
    let entityType: EntityType

    /// Desired position of new entity
    let position: Vec2Int

    /// Read BuildAction from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> BuildAction {
        var entityType: EntityType
        entityType = EntityType.readFrom(stream)
        var position: Vec2Int
        position = Vec2Int.readFrom(stream)
        return BuildAction(entityType: entityType, position: position)
    }

    /// Write BuildAction to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        entityType.writeTo(stream)
        position.writeTo(stream)
    }
}

File: generated-code/kotlin/codecraft/TcpReadWrite/src/main/kotlin/trans_gen_test/model/BuildAction.kt
package trans_gen_test.model

import trans_gen_test.util.StreamUtil

/**
 * Build action
 */
class BuildAction {
    /**
     * Type of an entity to build
     */
    var entityType: trans_gen_test.model.EntityType
    /**
     * Desired position of new entity
     */
    var position: trans_gen_test.Vec2Int

    constructor(entityType: trans_gen_test.model.EntityType, position: trans_gen_test.Vec2Int) {
        this.entityType = entityType
        this.position = position
    }

    /**
     * Write BuildAction to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, entityType.tag)
        position.writeTo(stream)
    }

    /**
     * Get string representation of BuildAction
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("BuildAction { ")
        stringBuilder.append("entityType: ")
        stringBuilder.append(entityType)
        stringBuilder.append(", ")
        stringBuilder.append("position: ")
        stringBuilder.append(position)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read BuildAction from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): BuildAction {
            var entityType: trans_gen_test.model.EntityType
            entityType = trans_gen_test.model.EntityType.readFrom(stream)
            var position: trans_gen_test.Vec2Int
            position = trans_gen_test.Vec2Int.readFrom(stream)
            return BuildAction(entityType, position)
        }
    }
}